[GNA] Update doc for GNA 3.5 (#18551)
openvinotoolkit/openvino: docs/OV_Runtime_UG/supported_plugins/GNA.md

- Prior to GNA 3.0, only 1D convolutions are natively supported on the HW; 2D convolutions have specific limitations (see the table below).
- The number of output channels for convolutions must be a multiple of 4.
- The maximum number of filters is 65532 for GNA 2.0 and 8192 for GNA 3.0.

> - Stating from The Intel® GNA 3.5 there was added support I8 weights to the convolutions. I8 Weights should be used with models after POT.

Reviewer: ```suggestion
- Stating from The Intel® GNA 3.5, support for int8 convolution weights was added. I8 Weights should be used with models after POT.
```

.. note:: 

>    The above limitations only apply to the new hardware 2D convolution operation. For GNA 3.0 When possible, the Intel® GNA

Reviewer: Is "the new operation" a correct term here, when we're describing older HW generation?

Also:
```suggestion
   The above limitations only apply to the new hardware 2D convolution operation. For GNA 3.0, when possible, the Intel® GNA
```

Reviewer 2: Convolution is new to GNA 3.x.

   plugin graph compiler flattens 2D convolutions so that the second generation Intel® GNA 1D convolution operations 
   (without these limitations) may be used. The plugin will also flatten 2D convolutions regardless of the sizes if GNA 2.0 
   compilation target is selected (see below).

Support for Convolutions since GNA 3.5
--------------------------------------------------------------------------------------------------------------------------------------

> Staring from The Intel® GNA 3.5 convolution 1D are handled in different way than in GNA 3.0. Convolution have following limitations:

Reviewer: ```suggestion
Staring from the Intel® GNA 3.5, 1D convolutions are handled in a different way than in the GNA 3.0. Convolutions have following limitations:
```

Reviewer 2: Adding "the" makes the English sound pretty awkward.  "GNA" is a name.  This is like saying "the Mike".  The other changes are OK. But PLEASE change "Staring" to "starting".

============================  =======================  =================
 Limitation                    Convolution 1D           Convolution 2D
============================  =======================  =================
Input height                   1                        1-65535
Input Width                    1-65535                  1-65535
Input channel number           1                        1-1024
Kernel number                  1-8192                   1-8192
Kernel height                  1                        1-255
Kernel width                   1-2048                   1-256
Stride height                  1                        1-255
Stride width                   1-2048                   1-256
Dilation height                1                        1
Dilation width                 1                        1
Pooling window height          1-1                      1-255
Pooling window width           1-255                    1-255
Pooling stride height          1                        1-255
Pooling stride width           1-255                    1-255
============================  =======================  =================

> Limitations for GNA 3.5 refers to the specific dimension. Whole scope of parameters is not fully supported,

Reviewer: ```suggestion
Limitations for GNA 3.5 refer to the specific dimension. Full range of parameters is not always supported,
```

Reviewer 2: Here is where "the" is needed.  "Full range of parameters" should be "The full range of parameters".

> e.g. where Convolution 2D Kernel can have height 255 and width 256, it may not work with Kernel with shape 255x256.

Reviewer: ```suggestion
e.g. even though Convolution 2D Kernel can have height 255 or width 256, a Kernel with shape 255x256 may not work.
```

Reviewer 3: I think it works better as in original
